src/Builder/Stage/UnsetStage.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * THIS FILE IS AUTO-GENERATED. ANY CHANGES WILL BE LOST!
+ */
+
+namespace MongoDB\Builder\Stage;
+
+use MongoDB\Builder\Encode;
+use MongoDB\Builder\Expression\FieldPath;
+use MongoDB\Builder\Type\StageInterface;
+
+/**
+ * Removes or excludes fields from documents.
+ * Alias for $project stage that removes or excludes fields.
+ *
+ * @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/unset/
+ */
+class UnsetStage implements StageInterface
+{
+    public const NAME = '$unset';
+    public const ENCODE = \MongoDB\Builder\Encode::Single;
+
+    /** @param list<FieldPath|non-empty-string> ...$field */
+    public array $field;
+
+    /**
+     * @param FieldPath|non-empty-string ...$field
+     * @no-named-arguments
+     */
+    public function __construct(FieldPath|string ...$field)
+    {
+        if (\count($field) < 1) {
```

Comment:
Is it expected that generated files reference global functions this way? It's inconsistent with what we do in the hand-written sources, so if it's easily configurable it may be worth changing.

Reply:
Added `$namespace->addUseFunction` calls to add this. The code generator don't know about the function body, so it cannot be detected.